query/command_graphite_test.go

```diff
@@ -0,0 +1,350 @@
+			expected: []api.Timeseries{
+				{
+					Values: []float64{5, 5, 5},
+					TagSet: api.TagSet{
+						"host":   "host68",
+						"method": "rpc",
+					},
+				},
+				{
+					Values: []float64{6, 6, 6},
+					TagSet: api.TagSet{
+						"host":   "host22",
+						"method": "rpc",
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		command, err := Parse(test.query)
+		if err != nil {
+			t.Fatalf("Unexpected error while parsing")
+			return
+		}
+		if command.Name() != "select" {
+			t.Errorf("Expected select command but got %s", command.Name())
+			continue
+		}
+		rawResult, err := command.Execute(ExecutionContext{Backend: fakeBackend, API: fakeAPI, FetchLimit: 1000, Timeout: 0})
+		if err != nil {
+			t.Errorf("Unexpected error while execution: %s", err.Error())
+			continue
+		}
+		seriesListList, ok := rawResult.([]api.SeriesList)
+		if !ok || len(seriesListList) != 1 {
+			t.Errorf("expected query `%s` to produce []value; got %+v :: %T", test.query, rawResult, rawResult)
+			continue
+		}
+		list := seriesListList[0]
+		if err != nil {
+			t.Fatal(err)
+		}
+		a := assert.New(t)
+		expectedSeries := test.expected
+		lookup := map[string][]float64{}
+		for _, expect := range expectedSeries {
+			lookup[expect.TagSet.Serialize()] = expect.Values
+		}
+		a.EqString(list.Name, "$graphite")
+		for _, series := range list.Series {
+			a.EqFloatArray(series.Values, lookup[series.TagSet.Serialize()], 1e-100)
+		}
+	}
+}
+
+// Verify that we filter before fetching
+func TestGraphiteLimits(t *testing.T) {
+	fakeAPI := fakeGraphiteAPI{}
+	fakeBackend := backend.NewSequentialMultiBackend(fakeGraphiteBackend{})
+	tests := []struct {
+		query    string
+		expected []api.Timeseries
+	}{
+		{
+			query: "graphite('%app%.%dc%.cpu.%quantity%') where app = 'server' and dc = 'north' from -10m to now",
+			expected: []api.Timeseries{
+				{
+					Values: []float64{1, 2, 3},
+					TagSet: api.TagSet{
+						"app":      "server",
+						"dc":       "north",
+						"quantity": "mean",
+					},
+				},
+				{
+					Values: []float64{4, 5, 6},
+					TagSet: api.TagSet{
+						"app":      "server",
+						"dc":       "north",
+						"quantity": "median",
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		command, err := Parse(test.query)
+		if err != nil {
+			t.Fatalf("Unexpected error while parsing")
+			return
+		}
+		if command.Name() != "select" {
+			t.Errorf("Expected select command but got %s", command.Name())
+			continue
+		}
+		rawResult, err := command.Execute(ExecutionContext{Backend: fakeBackend, API: fakeAPI, FetchLimit: 2, Timeout: 0})
+		if err != nil {
+			t.Errorf("Unexpected error while execution: %s", err.Error())
+			continue
+		}
+		seriesListList, ok := rawResult.([]api.SeriesList)
+		if !ok || len(seriesListList) != 1 {
+			t.Errorf("expected query `%s` to produce []value; got %+v :: %T", test.query, rawResult, rawResult)
+			continue
+		}
+		list := seriesListList[0]
+		if err != nil {
```

Comment:
Mistakenly left in from when another function was called; now removed in 37444f2